TACC/Core-Portal
# task/FP-86: Create common Sidebar

## Overview: ##
Created _common Sidebar react element and UIPatterns for it

## Related Jira tickets: ##

* [FP-86](https://jira.tacc.utexas.edu/browse/FP-86)

## Summary of Changes: ##
- Added new React component Sidebar to _common
- Wrote a unit test for new component
- Made UIPatterns entry for new component

## Testing Steps: ##
1. Navigate to UI Patterns and scroll to the bottom
2. Check that the Sidebar entry has similar behavior and styles as the other sidebars on the portal (Portal Sidebar, DataFiles, Applications, History, etc)

## UI Photos:

https://user-images.githubusercontent.com/43251554/158460340-dbda4f75-3e24-42da-bb1c-571389f78204.mov


## Notes: ##
